pkgs/by-name/as/asm-lsp/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, stdenv
+, darwin
+}:
+let
+  pname = "asm-lsp";
+  version = "0.4.2";
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "bergercookie";
+    repo = "asm-lsp";
+    rev = "v${version}";
+    hash = "sha256-EGgYOU6y23ULjnMGNjYhgF0JMPgvRuQ4UOWqwJxhBpU=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
```

Comment:
You should probably only import `Security` and have it inherited in the `callPackage` call.

Reply:
this is something nix-init does, I assume it has a purpose?

but looks like the program does not work on darwin anyway, so probably can be removed as a whole